.githooks/pre-commit

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+# Move to the git repository root
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd "$REPO_ROOT"
+
+# Stash non-staged changes to avoid linting them
+git stash push -q --keep-index
+
+# Run golangci-lint on all staged Go files
+echo "Running golangci-lint on staged Go files..."
+STAGED_GO_FILES=$(git diff --cached --name-only --diff-filter=ACM | grep '\.go$')
+if [ -n "$STAGED_GO_FILES" ]; then
+  # Use xargs to run golangci-lint on each file
+  echo "$STAGED_GO_FILES" | xargs golangci-lint run
```

Comment:
would it make sense to use our `just lint` command here instead of this?

Reply:
FYI you can probably just use `golangci-lint run --new-from-rev=XXX` here instead of all this.